src/state/slices/portfolioSlice/portfolioSlice.ts

```diff
@@ -26,18 +26,40 @@ export const portfolio = createSlice({
   initialState,
   reducers: {
     clear: () => initialState,
-    upsertPortfolio: (state, { payload }: { payload: Portfolio }) => {
+    upsertPortfolio: (state, { payload }: { payload: Portfolio & { basedOnNewTx?: boolean } }) => {
       // upsert all
       state.accounts.byId = { ...state.accounts.byId, ...payload.accounts.byId }
       const accountIds = Array.from(new Set([...state.accounts.ids, ...payload.accounts.ids]))
       state.accounts.ids = accountIds
-
-      state.assetBalances.byId = mergeWith(
-        state.assetBalances.byId,
-        payload.assetBalances.byId,
-        upsertAndSum
-      )
-
+      /**
+       * when refetching an account because of a new tx, we can't use the default
+       * upsertAndSum function otherwise the accounts balances will show false information.
+       * we instead got ot calculate the difference between the state accountBalance and the
+       * payload accountBalance for each of assets and then add [or subtract] the diff to the
+       * state.assetBalances item.
+       */
+      if (payload.basedOnNewTx) {
```

Comment:
i don't think we need this param, getAccount is only ever fetching one account. we should be able to do this logic every single time, and just comment as to why we're doing the logic (like you've done on 34-40 above)

Reply:
done